plenum/test/view_change/slow_nodes/test_view_change_gc_all_nodes_random_delay.py

```diff
@@ -0,0 +1,127 @@
+from itertools import combinations
+
+from stp_core.loop.eventually import eventually
+
+from plenum.test import waits
+from plenum.test.helper import sendRandomRequests, \
+        send_reqs_to_nodes_and_verify_all_replies, \
+        waitForSufficientRepliesForRequests, \
+        checkViewNoForNodes, \
+        check_last_ordered_3pc
+from plenum.test.delayers import delay_3pc_messages, \
+        reset_delays_and_process_delayeds
+from plenum.test.view_change.helper import ensure_view_change_complete
+from plenum.test.pool_transactions.conftest import clientAndWallet1, \
+        client1, wallet1, client1Connected, looper
+
+
+TestRunningTimeLimitSec = 300
+
+
+def ensure_nodes_last_ordered_3pc(nodes, last_ordered_3pc):
+    for n1, n2 in combinations(nodes, 2):
+        lst_3pc = check_last_ordered_3pc(n1, n2)
+
+    assert lst_3pc == last_ordered_3pc
+
+
+def ensure_nodes_requests_size(nodes, size):
+    for node in nodes:
+        assert len(node.requests) == size
+
+
+def test_view_change_gc_in_between_3pc_all_nodes_random_delays(
+        looper, txnPoolNodeSet, wallet1, client):
+    """
+    Test that garbage compares the whole 3PC key (viewNo, ppSeqNo) and
+    do not remove messages node's queues that have higher viewNo than
+    last ordered one even if their ppSeqNo are less or equal
+    """
+
+    numNodes = len(client.nodeReg)
+    viewNo = checkViewNoForNodes(txnPoolNodeSet)
+
+    # 1 send two messages one by one separately to make
+    #  node pool working with two batches
+    #    -> last_ordered_3pc = (+0, 2) [+0 means from the initial state]
+    #       (last_ordered_3pc here and futher is tracked
+    #       for master instances only cause non-master ones have
+    #       specific logic of its management which we don't care in
+    #       the test, see Replica::_setup_for_non_master)
+    send_reqs_to_nodes_and_verify_all_replies(looper, wallet1, client, 1)
+    send_reqs_to_nodes_and_verify_all_replies(looper, wallet1, client, 1)
+
+    last_ordered_3pc = (viewNo, 2)
+    ensure_nodes_last_ordered_3pc(txnPoolNodeSet, last_ordered_3pc)
```

Comment:
According to our convention methods starting with `ensure` use a looper and with for checks, this does not